Sources/Turf/Geometry.swift

```diff
@@ -4,21 +4,129 @@ import CoreLocation
 #endif
 
 
-public enum GeometryType: String {
+public enum GeometryType: String, Codable, CaseIterable {
     case Point
     case LineString
     case Polygon
     case MultiPoint
     case MultiLineString
     case MultiPolygon
+    case GeometryCollection
     
-    static let allValues: [GeometryType] = [.Point, .LineString, .Polygon, .MultiPoint, .MultiLineString, .MultiPolygon]
+//    static let allValues: [GeometryType] = [.Point, .LineString, .Polygon, .MultiPoint, .MultiLineString, .MultiPolygon]
 }
 
-public struct Geometry: Codable {
-    public var type: String
+public enum _Geometry {
+    private enum CodingKeys: String, CodingKey {
+        case type
+        case coordinates
+        case geometries
+    }
+    
+    case Point(coordinates: CLLocationCoordinate2D)
+    case LineString(coordinates: [CLLocationCoordinate2D])
+    case Polygon(coordinates: [[CLLocationCoordinate2D]])
+    case MultiPoint(coordinates: [CLLocationCoordinate2D])
+    case MultiLineString(coordinates: [[CLLocationCoordinate2D]])
+    case MultiPolygon(coordinates: [[[CLLocationCoordinate2D]]])
+    case GeometryCollection(geometries: [_Geometry])
+    
+    public var type: GeometryType {
+        switch self {
+        case .Point(_):
+            return .Point
+        case .LineString(_):
+            return .LineString
+        case .Polygon(_):
+            return .Polygon
+        case .MultiPoint(_):
+            return .MultiPoint
+        case .MultiLineString(_):
+            return .MultiLineString
+        case .MultiPolygon(_):
+            return .MultiPolygon
+        case .GeometryCollection(_):
+            return .GeometryCollection
+        }
+    }
     
-    public var geometryType: GeometryType? {
-        return GeometryType(rawValue: type)
+    public var value: Any? {
+        switch self {
+        case .Point(let value):
+            return value
+        case .LineString(let value):
+            return value
+        case .Polygon(let value):
+            return value
+        case .MultiPoint(let value):
+            return value
+        case .MultiLineString(let value):
+            return value
+        case .MultiPolygon(let value):
+            return value
+        case .GeometryCollection(let value):
+            return value
+        }
     }
 }
+
+extension _Geometry: Codable {
```

Comment:
Also, I see that some "Geometry" structs had additional methods (like searching for intersections, additional initializers and so on). To maintain these functionality, such methods will have to be moved inside this Enum or modified to become a free functions. How can I make this better than this?

Reply:
One possibility would be to have each case in the Geometry enumeration contain a struct that represents the actual geometry and can perform operations specific to that kind of geometry. For example, `case .Polygon(polygon: Polygon)`. Not sure if that would make things easier though.

In Turf.js, everything is a free function, but we originally wanted a more object-oriented API in Swift. Turf.js functions generally know how to return a meaningful result for any kind of geometry, so I think it could be OK to fold these methods into Geometry as long as we similarly generalize them.